code/go/0chain.net/smartcontract/dbs/event/snapshot.go

```diff
@@ -0,0 +1,59 @@
+package event
+
+import (
+	"errors"
+	"sync"
+
+	"0chain.net/core/logging"
+	"go.uber.org/zap"
+	"gorm.io/gorm"
+)
+
+// swagger:model Block
+type Snapshot struct {
+	gorm.Model
+	Round           int64  `gorm:"primaryKey;autoIncrement:false" json:"round"`
+	BlockHash       string `json:"block_hash"`
+	MintTotalAmount int64  `json:"mint_total_amount"`
+}
+
+var snapshotLock sync.Mutex
+
+func (edb *EventDb) GetRoundsMintTotal(from, to int64) (int64, error) {
+	var total int64
+	return total, edb.Store.Get().Model(&Snapshot{}).Where("round between ? and ?", from, to).Select("sum(mint_total_amount)").Scan(&total).Error
+}
+
+func (edb *EventDb) addOrUpdateTotalMint(mint Mint) error {
+	snapshotLock.Lock()
+	defer snapshotLock.Unlock()
+
+	snapshot, err := edb.getRoundSnapshot(mint.Round)
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			snapshot = &Snapshot{
+				Round:           mint.Round,
+				MintTotalAmount: mint.Amount,
+				BlockHash:       mint.BlockHash,
+			}
+			return edb.addSnapshot(snapshot)
```

Comment:
if I understand correctly you even need upsert, it can be done in postgres as follows INSERT ... ON CONFLICT (key) DO UPDATE https://stackoverflow.com/questions/17267417/how-to-upsert-merge-insert-on-duplicate-update-in-postgresql